nikola/shortcodes.py

```diff
@@ -83,6 +80,9 @@ def _skip_whitespace(data, pos, must_be_nontrivial=False):
 
 def _skip_nonwhitespace(data, pos):
     """Return first position not before pos which contains a non-whitespace character."""
+    for i, x in enumerate(data[pos:]):
+        if x.isspace():
+            return pos + i
```

Comment:
The `for` loop tries to find the first non-whitespace. If it doesn't find anything, the `while` loop (which is still there below) does the same thing afterwards, and it also won't find anything, so finally `len(data)` will be returned. You can replace the `while` loop with `return len(data)`.